pkg/systemd/bootInfo.jsx

```diff
@@ -0,0 +1,138 @@
+                    if (summaryElem !== null) {
+                        setSummary(summaryElem.textContent);
+                        summaryElem.remove();
+                    }
+                    const [plot, legend] = doc.querySelectorAll("g");
+                    legend.remove();
+                    [...plot.querySelectorAll("text.left"), ...plot.querySelectorAll("text.right")].forEach((text) => {
+                        const match = text.innerHTML.match(/^(?<service>.+\.[a-z._-]+)(\s+\((?<time>\d+(\.\d+)?)\w+\))?$/);
+                        if (match !== null) {
+                            const service = match.groups.service;
+                            const time = match.groups.time;
+                            text.setAttribute("data-service", service);
+                            text.setAttribute("data-time", time);
+                            text.classList.add("clickable-service");
+                        }
+                    });
+                    setSvg(doc.documentElement);
+                })
+                .catch((e) => {
+                    setSvg(null);
+                    setSummary(e);
+                });
+    }, []);
+
+    if (svg === undefined) {
+        return (
+            <div className="pf-v5-c-page__main-section">
+                <EmptyState variant={EmptyStateVariant.xs}>
+                    <EmptyStateHeader titleText={_("Loading")} headingLevel="h4" />
+                    <EmptyStateBody>
+                        <Spinner size="xl" />
+                    </EmptyStateBody>
+                </EmptyState>
+            </div>
+        );
+    }
+
+    if (svg === null) {
+        return (
+            <div className="pf-v5-c-page__main-section">
+                <EmptyState variant={EmptyStateVariant.xs}>
+                    <EmptyStateHeader titleText={_("Failure")} headingLevel="h4" />
+                    <EmptyStateBody>
+                        {_("Are you sure systemd-analyze is available?")}
+                        {_("systemd-analyze failed to load boot info and returned the following error:")}
+                        <CodeBlock>
+                            <CodeBlockCode id="code-content">{summary.toString()}</CodeBlockCode>
+                        </CodeBlock>
+                    </EmptyStateBody>
+                </EmptyState>
+            </div>
+        );
+    }
+
+    const plotClicked = (event) => {
+        const service = event.target.getAttribute("data-service");
+        if (service !== null) {
+            cockpit.jump(`/system/services#/${service}`, cockpit.transport.host);
+        }
+    };
+
+    return (
+        <div className="pf-v5-c-page__main-section">
+            <Card>
+                <CardTitle>{ _("Boot Info") }</CardTitle>
+                <CardBody>
+                    <p>
+                        {summary}
+                    </p>
+                    <List className="legend" isPlain variant={ListVariant.inline}>
```

Comment:
I don't understand how this works. This plus the CSS below define a set of custom colors and a legend, but the colors are already defined (and different) in the SVG produced by `systemd-analyze`. How do these tie together?

Reply:
The CSS and styles replace the colors used by systemd-analyze with the PatternFly color palette.